component/componenttest/configtest.go

```diff
@@ -37,17 +36,17 @@ func CheckConfigStruct(config any) error {
 // If the type is a struct it goes to each of its fields to check for the proper
 // tags.
 func validateConfigDataType(t reflect.Type) error {
-	var errs error
+	var errs []error
 
 	switch t.Kind() {
 	case reflect.Ptr:
-		errs = multierr.Append(errs, validateConfigDataType(t.Elem()))
+		errs = append(errs, validateConfigDataType(t.Elem()))
 	case reflect.Struct:
 		// Reflect on the pointed data and check each of its fields.
 		nf := t.NumField()
 		for i := 0; i < nf; i++ {
 			f := t.Field(i)
-			errs = multierr.Append(errs, checkStructFieldTags(f))
+			errs = append(errs, checkStructFieldTags(f))
```

Comment:
It doesn't matter that much here, but this adds lots of allocations. Do we care? Do we want to have a helper type (internal) "MultiError" with an append that does not append to the array if `nil`?

Reply:
That's a good point. I don't think we care in most places but there are some where we definitely do care; e.g. the fanoutconsumer `Consume*` calls seems like a place where we should avoid this.

Should we use the helper just for those places? We could also just continue using `multierr` on those for now and report this on golang/go

Reply:
> report this on golang/go 

I think we should definitely do that.

> Should we use the helper just for those places? We could also just continue using multierr on those for now and 

I think we can continue to use `multierr` for that.